cpp/src/gandiva/precompiled/string_ops_test.cc

```diff
@@ -1555,6 +1555,29 @@ TEST(TestStringOps, TestBinaryString) {
   EXPECT_EQ(output, "OM");
 }
 
+TEST(TestStringOps, TestFindInSet) {
+  gandiva::ExecutionContext ctx;
+  uint64_t ctx_ptr = reinterpret_cast<gdv_int64>(&ctx);
+
+  EXPECT_EQ(find_in_set_utf8_utf8(ctx_ptr, "HI", 2, "HI,B,C", 6), 1);
+  EXPECT_EQ(find_in_set_utf8_utf8(ctx_ptr, "HI", 2, ",B,C,HI", 7), 4);
+  EXPECT_EQ(find_in_set_utf8_utf8(ctx_ptr, "HI", 2, ",B,HI,HI,", 9), 3);
+  EXPECT_EQ(find_in_set_utf8_utf8(ctx_ptr, "", 0, ",B,A,A,", 7), 1);
+  EXPECT_EQ(find_in_set_utf8_utf8(ctx_ptr, "HI", 1, "HI,B,C", 6), 0);
+  EXPECT_EQ(find_in_set_utf8_utf8(ctx_ptr, "", 0, "B,C,A,", 6), 4);
+  EXPECT_EQ(find_in_set_utf8_utf8(ctx_ptr, "", 0, "B,C,,A,", 6), 3);
+  EXPECT_EQ(find_in_set_utf8_utf8(ctx_ptr, "A", 1, "B,A,", 4), 2);
+  EXPECT_EQ(find_in_set_utf8_utf8(ctx_ptr, "ha", 2, "hao,mn,hc,ha,hef", 16), 4);
+  EXPECT_EQ(find_in_set_utf8_utf8(ctx_ptr, "hef", 3, "hao,mn,hc,ha,hef", 16), 5);
```

Comment:
I think it would be good to add a test case in which the string that the method will search contains a comma:
```cpp
  EXPECT_EQ(find_in_set_utf8_utf8(ctx_ptr, "hao,", 2, "hao,mn,hc,ha,hef", 16), 4);
```

Reply:
Good point! Solved